File: smart-bank-modified/loan_engine.py

from data_handler import load_csv

def check_eligibility(customer_id):
    customers = load_csv('customers.csv')
    customer = customers[customers['customer_id'] == customer_id]
    if customer.empty: return "User not found"
    
    score = customer.iloc[0]['credit_score']
    if score > 750:
        return {"status": "High approval", "max_loan": "₹5,000,000", "interest": "8.5%"}
    elif 650 <= score <= 750:
        return {"status": "Medium", "max_loan": "₹1,000,000", "interest": "10.5%"}
    else:
        return {"status": "Low", "max_loan": "₹100,000", "interest": "14%"}

File: smart-bank-modified/server/loanEngine.ts

import { loadCSV } from './dataHandler';

export function getLoanEligibility(customerId: string) {
  const customers = loadCSV('customers.csv');
  const customer = customers.find((c: any) => c.customer_id === customerId);

  if (!customer) throw new Error("Customer not found");

  const score = parseInt((customer as any).credit_score || "0"); 
  let status = "Low";
  let maxLoan = 0;
  let interestRate = "12%";

  if (score > 750) {
    status = "High Approval Chance";
    maxLoan = 5000000;
    interestRate = "8.5%";
  } else if (score >= 650) {
    status = "Medium Approval Chance";
    maxLoan = 1000000;
    interestRate = "10.5%";
  } else {
    status = "Limited Eligibility";
    maxLoan = 100000;
    interestRate = "14%";
  }

  return {
    cibil_score: score,
    eligibility_status: status,
    max_loan_amount: `₹${maxLoan.toLocaleString('en-IN')}`,
    suggested_interest_rate: interestRate
  };
}
